lib/modules/datasource/azure-bicep-resource/index.ts

```diff
@@ -0,0 +1,74 @@
+import type { z } from 'zod';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import { BicepTypeIndex } from './schema';
+
+const BICEP_TYPES_INDEX_URL =
+  'https://raw.githubusercontent.com/Azure/bicep-types-az/main/generated/index.json';
+
+export class AzureBicepResourceDatasource extends Datasource {
+  static readonly id = 'azure-bicep-resource';
+
+  constructor() {
+    super(AzureBicepResourceDatasource.id);
+  }
+
+  async getReleases(
+    getReleasesConfig: GetReleasesConfig
+  ): Promise<ReleaseResult | null> {
+    const { packageName } = getReleasesConfig;
+
+    const resourceVersionIndex = await this.getResourceVersionIndex();
+    const versions = resourceVersionIndex.get(packageName.toLowerCase());
+
+    if (!versions) {
+      return null;
+    }
+
+    const firstSlashIndex = packageName.indexOf('/');
+    const namespaceProvider = packageName
+      .slice(0, firstSlashIndex)
+      .toLowerCase();
+    const type = packageName.slice(firstSlashIndex + 1).toLowerCase();
+
+    const changeLogUrl = `https://learn.microsoft.com/en-us/azure/templates/${namespaceProvider}/change-log/${type}`;
+
+    return {
+      releases: versions.map((x) => ({
+        version: x,
+        changeLogUrl, // TODO: does not show up in PR yet?
+      })),
+    };
+  }
+
+  async getResourceVersionIndex(): Promise<Map<string, string[]>> {
```

Comment:
add cache decorators, checkout other datasources.

make helper functions private

Reply:
I added the cache decorators, but I could not make the `getResourceVersionIndex` method private. I think the cache decorator only works for non-private methods. as it yields this error: `Argument of type '"getResourceVersionIndex"' is not assignable to parameter of type 'keyof AzureBicepResourceDatasource'.ts(2345)`. Changing the cache key `getReleases` (non-private method) did not help.

Reply:
Yeah, `@cache` requires public functions.